creusot/src/backend/program.rs

```diff
@@ -148,56 +147,84 @@ pub(crate) fn translate_function<'tcx, 'sess>(
     let tcx = ctx.tcx;
     let mut names = Dependencies::new(tcx, [def_id]);
 
-    let Some(body_ids) = collect_body_ids(ctx, def_id) else {
+    let Some((body_id, promoteds)) = collect_body_ids(ctx, def_id) else {
         let (_, clones) = names.provide_deps(ctx, GraphDepth::Deep);
         return (clones, None);
     };
-    let body = to_why(ctx, &mut names, body_ids[0]);
+    let body = Decl::Coma(to_why(ctx, &mut names, body_id));
 
     if ctx.tcx.is_closure_like(def_id) {
         closure_aux_defs(ctx, def_id)
     };
 
-    let promoteds =
-        body_ids[1..].iter().map(|body_id| to_why(ctx, &mut names, *body_id)).collect::<Vec<_>>();
+    let promoteds = promoteds
+        .iter()
+        .map(|body_id| Decl::Coma(to_why(ctx, &mut names, *body_id)))
+        .collect::<Vec<_>>();
+
+    let tcx = ctx.tcx;
+    let ghost_closure_ids: IndexSet<_> = names.find_ghost_closures(tcx).collect();
+    let ghost_closures = ghost_closure_ids
+        .into_iter()
+        .map(|(def_id, subst)| {
+            // Compile the closure normally
+            closure_aux_defs(ctx, def_id);
```

Comment:
this only is required if ghost closures are not compiled ordinarily, is that still the case?

Reply:
Indeed this is not the case, this is now removed